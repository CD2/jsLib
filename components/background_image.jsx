import React from 'react'
import PropTypes from 'prop-types'
import { styled, p } from 'utils/theme'
import decorate from 'utils/decorate'

export class BackgroundImage extends React.Component {

  static PropTypes = {
    src: PropTypes.string,
    default_src: PropTypes.string,
    width: PropTypes.string,
    height: PropTypes.string,
    contain: PropTypes.bool,
    circular: PropTypes.bool,
  }

  render() {
    const { className, children } = this.props

    return (
      <div className={className}>
        { children }
      </div>
    )
  }

}
export default decorate(
  styled`
    width: ${p('width', '100%')};
    height: ${p('height', '100%')};
    ${props => props.circular ? `border-radius: 50%;` : ``}
    ${({ src, default_src, contain }) => {
      const image = src || default_src
      const backgroundSize = contain ? 'contain' : 'cover'
      if (image) {
          return `
            background-size: ${backgroundSize};
            background-image: url(${image});
            background-position: 50%;
            background-repeat: no-repeat;
          `
        } else {
          return `background-color: #ddd;`
      }
    }

    }
  `,
  BackgroundImage
)
