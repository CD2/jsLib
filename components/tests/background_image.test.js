import t from 'utils/test_helper'

import { BackgroundImage } from '../background_image'

t.describe_component(BackgroundImage, {}, (c) => {

})
