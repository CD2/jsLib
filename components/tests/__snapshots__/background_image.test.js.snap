// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <BackgroundImage /> renders correctly 1`] = `"<div></div>"`;
